mkdocs_simple_plugin/plugin.py

```diff
@@ -106,23 +243,46 @@ def copy_doc_files(self, dest_dir):
         for root, dirs, files in os.walk("."):
             if self.in_include_dir(root):
                 for f in files:
-                    if self.in_extensions(f):
-                        doc_root = dest_dir + root[1:]
-                        orig = "{}/{}".format(root, f)
-                        new = "{}/{}".format(doc_root, f)
-                        try:
-                            os.makedirs(doc_root, exist_ok=True)
-                            shutil.copy(orig, new)
-                            utils.log.debug(
-                                "mkdocs-simple-plugin: {} --> {}".format(orig, new))
-                            paths.append((orig, new))
-                        except Exception as e:
-                            utils.log.warn(
-                                "mkdocs-simple-plugin: error! {}.. skipping {}".format(e, orig))
-
+                    paths.extend(self.copy_extract(root, f, dest_dir))
             dirs[:] = [d for d in dirs if self.in_search_dir(d, root)]
         return paths
 
+    def copy_extract(self, from_dir, f, dest_dir):
```

Comment:
Prefer to break this into two functions -- copy and extract

Reply:
OK, those two operations share a bunch of temporary local variable context which is why I lumped them, but I will just find the cleanest way to split them up in the revised PR.